### ISSUE TITLE
How to add custom data to the session?

### ISSUE THREAD

**Issue by Unbuttun** [mgcrea/fastify-session#12]
For example, while I was using express-session or the other fastify-session plugin, i was able to add custom data by doing `req.session.userId='some id'` after extending the request object.

I've been trying similar process here by doing something like this but the custom session data isn't getting saved. (as commented in this file - https://github.com/mgcrea/fastify-session/blob/master/src/session/SessionData.ts)

```js
declare module "fastify" {
  interface FastifyRequest {
    session: Session;
  }
}

declare module "@mgcrea/fastify-session" {
  interface SessionData {
    userId: string;
  }
}
```

Adding it like this - 
```js
    await app.register(fastifyCookie);
    await app.register(fastifySession, {
      key: Buffer.from(SESSION_KEY, "base64"),
      crypto: SODIUM_AUTH,
      store: new RedisStore({ client: new Redis(REDIS_URI), ttl: SESSION_TTL }),
      cookie: { maxAge: SESSION_TTL }
    });
```
then using it here - 
```js
    app.get("/set", async (request: FastifyRequest) => {
      request.session.data.userId = "1";
      const id = request.session.data.userId;
      console.log("ID", id);
      return {
        id,
      };
    });

    app.get("/get", async (request: FastifyRequest) => {
      const id = request.session.data.userId;
      return {
        id,
      };
    });
```

And the custom data that i set does not stay persistent. I've looked into the source code but I'm not exactly able to figure out this issue. The sessionId remains same in both request when i console them.

Thanks.

**Comment by Unbuttun** [comment]
nvm, i should have read the docs clearly.

TLDR being, rather than using the original fastify module to start the server, I needed to use the buildFastify function that was told in the readme to do that.

```js
  const app = await buildFastify({
    logger: {
      prettyPrint: true
    }
  })
  ```
 something like this.
